src/electrum.rs

```diff
@@ -275,6 +275,88 @@ impl Rpc {
         Ok(status)
     }
 
+    fn outpoint_subscribe(&self, (txid, vout): (Txid, u32)) -> Result<Value> {
+        let funding = OutPoint { txid, vout };
+
+        let funding_blockhash = self.tracker.get_blockhash_by_txid(funding.txid);
+        let spending_blockhash = self.tracker.get_blockhash_spending_by_outpoint(funding);
+
+        let mut inputs_funding_confirmed = true;
+
+        let funding_tx = self.daemon.get_transaction(&funding.txid, funding_blockhash);
+
+        if funding_tx.is_err() {
+            return Ok(json!({}));
+        }
+
+        let funding_inputs = &funding_tx.as_ref().unwrap().input;
+        let outpoint_script = &funding_tx.as_ref().unwrap().output.get(vout as usize).unwrap().script_pubkey;
+
+        if funding_blockhash.is_none() && funding_inputs.iter().any(|txi| self.tracker.get_blockhash_by_txid(txi.previous_output.txid).is_none()) {
+            inputs_funding_confirmed = false;
+        }
+
+
+        let scripthash = ScriptHash::new(&outpoint_script);
+        let outpoint_scripthash_status = self.new_status(scripthash);
+
+        let funding_height = self.tracker.chain().get_block_height(&funding_blockhash.unwrap()).unwrap();
```

Comment:
This will crash if funding transaction i unconfirmed -another DoS vector. I suggest this:

```rust
let funding_height = match &funding_blockhash {
    Some(funding_blockhash) => self.tracker.chain().get_block_height(funding_blockhash)?,
    None => -1,
};
```

Reply:
I have to return 0 in the None case or the compiler complain that Neg is not implemented for usize. This is fine :) thanks for the catch !